frontend/utils/distribution/group-income-distribution.js

```diff
@@ -1,40 +1,7 @@
-import { saferFloat } from '~/frontend/views/utils/currencies.js'
-import incomeDistribution from '~/frontend/utils/distribution/mincome-proportional.js'
+import parseMonthlyDistributionFromEvents from '~/frontend/utils/distribution/monthly-event-parser.js'
+import minimizeTotalPaymentsCount from '~/frontend/utils/distribution/payments-minimizer.js'
 
-export default function groupIncomeDistribution ({ state, getters, monthstamp, adjusted }: Object): any {
-  // the monthstamp will always be for the current month. the alternative
-  // is to allow the re-generation of the distribution for previous months,
-  // but that approach requires also storing the historical mincomeAmount
-  // and historical groupProfiles. Since together these change across multiple
-  // locations in the code, it involves less 'code smell' to do it this way.
-  // see historical/group.js for the ugly way of doing it.
-  const mincomeAmount = getters.groupMincomeAmount
-  const groupProfiles = getters.groupProfiles
-  const currentIncomeDistribution = []
-  for (const username in groupProfiles) {
-    const profile = groupProfiles[username]
-    const incomeDetailsType = profile && profile.incomeDetailsType
-    if (incomeDetailsType) {
-      const adjustment = incomeDetailsType === 'incomeAmount' ? 0 : mincomeAmount
-      const amount = adjustment + profile[incomeDetailsType]
-      currentIncomeDistribution.push({
-        name: username,
-        amount: saferFloat(amount)
-      })
-    }
-  }
-  let dist = incomeDistribution(currentIncomeDistribution, mincomeAmount)
-  if (adjusted) {
-    // if this user has already made some payments to other users this
-    // month, we need to take that into account and adjust the distribution.
-    // this will be used by the Payments page to tell how much still
-    // needs to be paid (if it was a partial payment).
-    for (const p of dist) {
-      const alreadyPaid = getters.paymentTotalFromUserToUser(p.from, p.to, monthstamp)
-      // if we "overpaid" because we sent late payments, remove us from consideration
-      p.amount = saferFloat(Math.max(0, p.amount - alreadyPaid))
-    }
-    dist = dist.filter(p => p.amount > 0)
-  }
-  return dist
+export default function groupIncomeDistribution (distributionEvents: Array<Object>, { mincomeAmount, adjusted = false, minimizeTxns = false, monthlyRated = true }: Object): any {
+  const dist = parseMonthlyDistributionFromEvents(distributionEvents, mincomeAmount, monthlyRated, adjusted)
```

Comment:
See this discussion for how this function should look (it should call two different functions, based on the value of `options.adjusted`): https://github.com/okTurtles/group-income-simple/pull/1056#discussion_r603473612

Reply:
After discussion with @CrazedCoding I think we possibly came to an agreement, that we're going to leave the handling of the `options.adjusted` parameter inside of the `parseMonthlyDistributionFromEvents` function, so that if we ever enable pro-rating, we can get an unadjustedDistribution that accounts for the pro-rating of the incomes.